mipengine/controller/node_landscape_aggregator.py

```diff
@@ -0,0 +1,337 @@
+        )
+        for data_model, datasets in datasets_per_data_model.items():
+            if data_model not in datasets_locations:
+                datasets_locations[data_model] = {dataset: [] for dataset in datasets}
+            for dataset in datasets:
+                datasets_locations[data_model][dataset].append(node_info.id)
+    return datasets_locations
+
+
+async def _get_datasets_labels(nodes: List[NodeInfo]) -> Dict[str, Dict[str, str]]:
+    datasets_labels = {}
+    for node_info in nodes:
+        node_socket_addr = _get_node_socket_addr(node_info)
+        datasets_per_data_model = await _get_node_datasets_per_data_model(
+            node_socket_addr
+        )
+        for data_model, datasets in datasets_per_data_model.items():
+            datasets_labels[data_model] = {}
+            for dataset in datasets:
+                datasets_labels[data_model][dataset] = datasets[dataset]
+    return datasets_labels
+
+
+async def _get_cdes_across_nodes(
+    nodes: List[NodeInfo],
+) -> Dict[str, List[Tuple[str, CommonDataElements]]]:
+    nodes_cdes = {}
+    for node_info in nodes:
+        node_socket_addr = _get_node_socket_addr(node_info)
+        datasets_per_data_model = await _get_node_datasets_per_data_model(
+            node_socket_addr
+        )
+        for data_model in datasets_per_data_model:
+            cdes = await _get_node_cdes(node_socket_addr, data_model)
+            if data_model not in nodes_cdes:
+                nodes_cdes[data_model] = []
+            nodes_cdes[data_model].append((node_info.id, cdes))
+    return nodes_cdes
+
+
+def _get_compatible_data_models(
+    data_model_cdes_across_nodes: Dict[str, List[Tuple[str, CommonDataElements]]]
+) -> Dict[str, CommonDataElements]:
+    """
+    Each node has its own data models definition.
+    We need to check for each data model if the definitions across all nodes is the same.
+    If the data model is not the same across all nodes containing it, we log the incompatibility.
+    The data models with similar definitions across all nodes are returned.
+
+    Parameters
+    ----------
+        data_model_cdes_across_nodes: the data_models each node has
+
+    Returns
+    ----------
+        Dict[str, CommonDataElements]
+            the data models with similar definitions across all nodes
+
+    """
+    data_models = {}
+    for data_model, cdes_from_all_nodes in data_model_cdes_across_nodes.items():
+        first_node, first_cdes = cdes_from_all_nodes[0]
+        for node, cdes in cdes_from_all_nodes[1:]:
+            if not first_cdes == cdes:
+                logger.info(
+                    f"Node '{first_node}' and node '{node}' on data model '{data_model}' have incompatibility on the following cdes: {first_cdes} and {cdes}"
+                )
+                break
+        else:
+            data_models[data_model] = first_cdes
+
+    return data_models
+
+
+def get_updated_data_model_with_dataset_enumerations(
```

Comment:
minor, but what makes this data model "updated" instead of just data model?

Reply:
"Updated" this data model will contain the a new dataset enumerations, so the "updated" is to  inform that the cdes of the data model will be modified